### PR TITLE
Allow adding or updating an invoice PO number

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Allow adding or updating an invoice PO number

[stage-2]

## Motivation and Context
Billing Frictions epic

## How Has This Been Tested?
Tested changes locally. Will update unit tests in a subsequent PR.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Initial PR for this, mainly to review the behaviour. Will update unit tests and any suggestions re: functionality separately.

@ezequielc @santiagonoguez please review. Thanks!

**Comment by santiagonoguez** [comment]
@alex-deaconu how can I get to this invoice page in stage 2 ? Thanks !

**Comment by ezequielc** [comment]
@santiagonoguez You can find an unpaid invoices in the Billing page, like this: https://apps-stage-2.risevision.com/billing/invoice/6806?token=estKey&cid=b428b4e8-c8b9-41d5-8a10-b4193c789443. Thanks!

File: web/scripts/billing/services/svc-billing.js [hunk 1]
```diff
@@ -107,6 +111,34 @@ angular.module('risevision.apps.billing.services')
 
           return deferred.promise;
         },
+        updateInvoice: function (invoice, companyId, token) {
+          var deferred = $q.defer();
+
+          var fields = pick.apply(this, [invoice].concat(INVOICE_WRITABLE_FIELDS));
```
Comment (santiagonoguez) [inline web/scripts/billing/services/svc-billing.js:117]: Curious, I'm not familiar with this pick utility. I assume it's something for populating data, or where I can find how does it work ? Thanks !

File: web/scripts/billing/controllers/ctr-invoice.js [hunk 2]
```diff
@@ -22,5 +22,19 @@ angular.module('risevision.apps.billing.controllers')
...
+        billingFactory.invoice.poNumber = billingFactory.invoice.poNumber || '';
+
+        billingFactory.updatePoNumber()
+          .then(function() {
+            $scope.clearPoNumber();
+          });
+      };
+
+      $scope.clearPoNumber = function () {
+        billingFactory.invoice.poNumber = null;
```
Comment (ezequielc) [inline web/scripts/billing/controllers/ctr-invoice.js:35]: @alex-deaconu Could you clarify why you need to set `null` here? I tested and works fine, but I am a bit confused about the behaviour here. Thanks!

**Review by ezequielc — APPROVED** [review]

File: web/scripts/billing/services/svc-billing.js (hunk 1 above)
Comment (alex-deaconu) [reply web/scripts/billing/services/svc-billing.js:117]: @santiagonoguez yes, it erases all properties of an object other than the specific ones. Util is here https://github.com/Rise-Vision/rise-vision-apps/blob/master/web/scripts/components/core-api-client/svc-core-util.js#L6.

It's used in all our API code that's why I just re-used it, for example: https://github.com/Rise-Vision/rise-vision-apps/blob/master/web/scripts/displays/services/svc-display.js#L113.

Thanks.

File: web/scripts/billing/controllers/ctr-invoice.js (hunk 2 above)
Comment (alex-deaconu) [reply web/scripts/billing/controllers/ctr-invoice.js:35]: @ezequielc good catch. It should probably not be there, I will remove. Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc @santiagonoguez I've switched the show/hide mechanism for the textbox to use `ng-if`. This way the value is cleared to use the `ng-value` when it's hidden and then shown. Thanks.

**Pull request merged**